File: Binary Search/codeforces_379072E.py

# Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/E
import sys
import bisect


def main():
    # Read the number of items in the price list
    n = int(sys.stdin.readline().strip())

    # Read the price list and sort it
    price_list = list(map(int, sys.stdin.readline().split()))
    price_list.sort()

    # Read the number of queries
    q = int(sys.stdin.readline().strip())

    # Process each query
    for _ in range(q):
        m = int(sys.stdin.readline().strip())  # Read the price limit

        # Find the number of items that cost less than or equal to m
        count = bisect.bisect_right(price_list, m)

        # Print the result
        print(count)


if __name__ == "__main__":
    main()

File: Binary Search/codeforces_379072E.cpp

// Problem link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/problem/E
// submission link ===>> https://codeforces.com/group/isP4JMZTix/contest/379072/submission/312650360

#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define endl '\n'

int main()
{
    ios_base::sync_with_stdio(false);
    cout.tie(nullptr);
    cin.tie(nullptr);

    ll n;
    cin >> n;
    vector<int> priceList;
    for (int i = 0, x; i < n; i++)
    {
        cin >> x;
        priceList.push_back(x);
    }
    sort(priceList.begin(), priceList.end());
    int q;
    cin >> q;
    while (q--)
    {
        int m;
        cin >> m;
        cout << upper_bound(priceList.begin(), priceList.end(), m) - priceList.begin() << endl;
    }
    return 0;
}